internal/shardchain/leveldb_shard/shard.go

```diff
@@ -0,0 +1,200 @@
+package leveldb_shard
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/comparer"
+	"github.com/syndtr/goleveldb/leveldb/filter"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/util"
+)
+
+type LeveldbShard struct {
+	dbs     []*leveldb.DB
+	dbCount uint32
+}
+
+var shardIdxKey = []byte("__DB_SHARED_INDEX__")
+
+func NewLeveldbShard(savePath string, diskCount int, diskShards int) (shard *LeveldbShard, err error) {
+	shard = &LeveldbShard{
+		dbs:     make([]*leveldb.DB, diskCount*diskShards),
+		dbCount: uint32(diskCount * diskShards),
+	}
+
+	// clean when error
+	defer func() {
+		if err != nil {
+			for _, db := range shard.dbs {
+				if db != nil {
+					_ = db.Close()
+				}
+			}
+
+			shard = nil
+		}
+	}()
+
+	levelDBOptions := &opt.Options{
```

Comment:
make those numbers configurable with default values

Reply:
My suggestion is to use those fixed value, These parameters have been tested. Too large or too small configurable will affect memory usage and performance.